src/traces/parcoords/attributes.js

```diff
@@ -80,8 +80,22 @@ module.exports = {
             valType: 'info_array',
             role: 'info',
             items: [
-                {valType: 'number', editType: 'calc'},
-                {valType: 'number', editType: 'calc'}
+                {
```

Comment:
This just shows the current structure, there won't be a limit of two specified intervals.

Reply:
(for my own notes, @monfera don't worry about this unless it's blocking you)

If we merge in master, particularly #2399, we can use 2D and/or unbounded free-length `info_array`.
I'm not sure the best way to handle it though - since the vast majority of cases will still use a single range, I'd like to keep `constraintrange` 1D rather than forcing people to wrap it into 2D. So the two options that occur to me are:
1. make it a length-2N array, ie `[min1, max1, min2, max2, min3, max3...]` - this would be fairly simple for us to work with, and seems easy to *describe* to users, not sure how intuitive it will be for them to use though.
2. make a separate 2D attribute `constraintranges` that we switch to when there are multiple ranges. That would be more plumbing on our side, and we'd have to decide which one wins in a conflict, but the structure may be more intuitive.

@etpinard thoughts?

Reply:
Users should **not** have to make `constraintrange` 2D to set a single constraint. 

I'll like to add a third option, make `constraintrange` 1D or 2D. This would be a first for info arrays, but note that some data arrays already use this pattern (e.g. heatmap `x` and `y`).

Reply:
Another aspect is that the functionality is [initially discussed](https://github.com/plotly/plotly.js/issues/2331#issuecomment-364920486) specifically for ordinal dimensions, where it'd be natural to simply list values, eg. `[1, 5, 8, 12]`, and even to mix singular values and ranges, like `[1, [5, 8], 12]` meaning 1, 5 to 8 and 12. It would be possible to support this, but it wouldn't blend well with using an 1D array for a single range, because then an 1D array with two elements could be interpreted as either a single range, or as two discrete values.